src/routes/chat_route.py

```diff
@@ -192,7 +189,47 @@ async def add_user_to_agreed():
                 for user in chat.users:
                     print(f"{user} received {chat.words} $WORD")
                     user.words += chat.words
-                # Save chat text to ChatFullText
+
+                # Check if the user is looged in and has an open session
+                user_db_session = await get_db_session(None, user.id)
+                if user_db_session is None:
+                    print(
+                        "Session does not exist: Chat is sold, but has not been fetched!"
+                    )
+                    # we still return 200, because the chat is sold
+                    continue
+
+                client = TelegramClient(
+                    StringSession(user_db_session.id), API_ID, API_HASH
+                )
+                if await connect_client(client, None, user.id) == -1:
+                    print("Error in connecting to Telegram")
+                    # we still return 200, because the chat is sold
+                    continue
+                if not await client.is_user_authorized():
```

Comment:
We have to disconnect from Telegram after each successful connect. Maybe add it under `if not await client.is_user_authorized():` so we always disconnect